### PR TITLE
Update getting started

### PR DESCRIPTION
**Pull request by DanPaseltiner**
This PR updates the getting started doc.

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -1,232 +1,154 @@
...
+| Ingestion | Provide endpoints for each step of the ingestion pipeline except conversion to FHIR. | [https://github.com/CDCgov/phdi/tree/main/containers/ingestion](https://github.com/CDCgov/phdi/tree/main/containers/ingestion) |
+| Tabulation | Extract and tabularize data from the FHIR server according to user-defined schema. |  [https://github.com/CDCgov/phdi/tree/main/containers/tabulation](https://github.com/CDCgov/phdi/tree/main/containers/tabulation) |
 
-Under the hood, this integration is doing the following:
+### Azure Project Configuration
 
-1. Creates a virtual environment (default path `.venv`) and installing all depedencies called out in `requirements.txt`. You can alternatively do this yourself with `source .venv/bin/activate; pip install -r requirements.txt`.
-2. Creates `.vscode/tasks.json`, which make it easier to activate the relevant virtual environment, installs dependencies, and starts a debugging session
-3. Creates `.vscode/launch.json`, which makes it so that when you hit F5 / go to `Run->Start Debugging` it runs the tasks from (2) and then attaches to the debugger for your functions when running locally.
+Unknown - TODO??
```
Comment (gordonfarrell) [inline docs/getting_started.md:101]: This TODO meant to be left in?

File: docs/getting_started.md [hunk 2]
```diff
@@ -1,232 +1,154 @@
...
+### Azure Data Factory (ADF): Orchestration
+We use [Azure Data Factory (ADF)](https://learn.microsoft.com/en-us/azure/data-factory/control-flow-azure-function-activity) to define processes that require the use of multiple Building Blocks. These data flows can be developed using [The Azure Function Activity UI](https://learn.microsoft.com/en-us/azure/data-factory/control-flow-azure-function-activity#create-an-azure-function-activity-with-ui). However for best Infrastructure as Code (IaC) best practices we recommend deploying ADF pipelines in production from JSON configuration files. This allows consistency and reproducibility between deployments and the use of source control to track changes to the pipline over time. The JSON configuration file for the ingestion pipeline is available in [`/terraform/modules/data_factory/ingestion-pipeline.json`](../terraform/modules/data_factory/ingestion-pipeline.json) in this repository.
```
Comment (gordonfarrell) [inline docs/getting_started.md:72]: Double best: "However for best Infrastructure as Code (IaC) best practices we recommend"

**Review by gordonfarrell — APPROVED** [review]
Looks good just a couple tiny things

File: docs/getting_started.md [hunk 3]
```diff
@@ -1,232 +1,154 @@
...
...e/healthcare-apis/azure-api-for-fhir/overview)). Each step of the pipeline, orchestrated by ADF, is simply a REST API call to indepent containerized microservices (AKA Building Blocks) developed by PHDI ([available here](https://github.com/orgs/CDCgov/packages?repo_name=phdi)). These services are deployed in Azure as [Azure Container Apps (ACA)](https://learn.microsoft.com/en-us/azure/container-apps/). Each endpoint on a service performs a single function, or step in a pipeline (e.g patient name standardization), and returns the processed data back to the data factory where it is passed on to the next service. Endpoints that perform related functions (e.g. name standardization and phone number standardization) are provided by a single service. Ingested data may be accessed via the FHIR server's API ([HL7 docs](https://hl7.org/fhir/http.html), [Azure docs](https://learn.microsoft.com/en-us/azure/healthcare-apis/azure-api-for-fhir/overview-of-search)), or the tabulation service. When called, the tabulation service with extract data, in tabular form according to a user-defined schema, from the FHIR server and writen to flat files. This entire architecture is show in the diagram below.
```
Comment (nickclyde) [inline docs/getting_started.md:31]:
```suggestion
Source data is stored in Azure using [Azure Blob Storage](https://learn.microsoft.com/en-us/azure/storage/blobs/). When new data is written to this blob store the `read-source-data` [Azure Function](https://learn.microsoft.com/en-us/azure/data-factory/control-flow-azure-function-activity) is triggered. This function simply reads the new data from blob storage and passes it, along with some metadata, to the ingestion pipeline for further processing. The ingestion pipeline is implemented in [Azure Data Factory (ADF)](https://learn.microsoft.com/en-us/azure/data-factory/) and is responsible for data cleaning, standardization, enrichment, record linkage, and finally uploading data to a FHIR server ([Azure API for FHIR](https://learn.microsoft.com/en-us/azure/healthcare-apis/azure-api-for-fhir/overview)). Each step of the pipeline, orchestrated by ADF, is simply a REST API call to independent containerized microservices (AKA Building Blocks) developed by PHDI ([available here](https://github.com/orgs/CDCgov/packages?repo_name=phdi)). These services are deployed in Azure as [Azure Container Apps (ACA)](https://learn.microsoft.com/en-us/azure/container-apps/). Each endpoint on a service performs a single function or step in a pipeline (e.g patient name standardization), and returns the processed data back to the data factor. It is then passed on to the next service. Endpoints that perform related functions (e.g. name standardization and phone number standardization) are provided by a single service. Ingested data may be accessed via the FHIR server's API ([HL7 docs](https://hl7.org/fhir/http.html), [Azure docs](https://learn.microsoft.com/en-us/azure/healthcare-apis/azure-api-for-fhir/overview-of-search)), or the tabulation service. When called, the tabulation service will extract data, in tabular form according to a user-defined schema, from the FHIR server and writen to flat files. This entire architecture is shown in the diagram below.
```

Comment (nickclyde) [reply docs/getting_started.md:31]: (Fixed a few typos and punctuation issues)

File: docs/getting_started.md [hunk 4]
```diff
@@ -1,232 +1,154 @@
...
+1. `source-data`
+2. `fhir-conversion-failures`
+3. `fhir-upload-failures`
 
-You can easily download the environment variable configuration for a given function app using the azure CLI with:
+Data in the form it is originally received by the starter kit is stored in the `source-data`. In practice the pipeline has two failure modes, on occasion issues will be encountered converting source data to FHIR at the beginning of the ingestion pipeline and uploading data to the FHIR server at the end of the ingestion pipeline. In the event of FHIR conversion and upload failures the data that could not be converted or uploaded is written to the appropriate container (`fhir-conversion-failures`, or `fhir-upload-failures`) as a JSON blob along with logging describing the specific failure. Additionally, each of these three containers contains a subdirectory for each type of public health data the starter kit can process. A new ELR message for the pipeline to ingest would be written to `source-data/elr/` while an eCR message that failed to be uploaded to the FHIR server would be written to `fhir-upload-failures/ecr`. The overall directory structure for the PHI storage account is shown below.
```
Comment (nickclyde) [inline docs/getting_started.md:47]:
```suggestion
Data in the form it is originally received by the starter kit is stored in the `source-data`. In practice the pipeline has two failure modes. On occasion, issues will be encountered converting source data to FHIR at the beginning of the ingestion pipeline and uploading data to the FHIR server at the end of the ingestion pipeline. In the event of FHIR conversion and upload failures, the data that could not be converted or uploaded is written to the appropriate container (`fhir-conversion-failures`, or `fhir-upload-failures`) as a JSON blob along with logging describing the specific failure. Additionally, each of these three containers contains a subdirectory for each type of public health data the starter kit can process. A new ELR message for the pipeline to ingest would be written to `source-data/elr/` while an eCR message that failed to be uploaded to the FHIR server would be written to `fhir-upload-failures/ecr`. The overall directory structure for the PHI storage account is shown below.
```

File: docs/getting_started.md [hunk 5]
```diff
@@ -1,232 +1,154 @@
...
-### Hardware
-
-Until we have properly containerized our apps, we will need to rely on informal consensus around hardware. Here is a list of machines that are compatible with development:
-- Intel Macs
-- Apple-Silicon Macs
-- Windows-based machines with Windows 10/11 Home or higher. However, as the work moves towards containerization, Windows Pro will be necessary in order to run Docker.
-
-### Software
+#### Azure Functions Storage Account
+The second storage account in the starter kit is the Azure Functions Storage Account. This account is used for all storage needs related to Azure Functions. The starter kit currently creates a containers in this account to store the source code for the [`read-source-data](#azure-functions) function.
```
Comment (nickclyde) [inline docs/getting_started.md:66]:
```suggestion
The second storage account in the starter kit is the Azure Functions Storage Account. This account is used for all storage needs related to Azure Functions. The starter kit currently creates a container in this account to store the source code for the [`read-source-data](#azure-functions) function.
```

File: docs/getting_started.md [hunk 6]
```diff
@@ -1,232 +1,154 @@
...
 
-#### Overview
-The team uses VSCode as its IDE, but other options (e.g. IntelliJ, Eclipse, PyCharm, etc.) can be viable as well. The main drivers behind using VSCode are its integration with Azure, which is a natural byproduct of them both being Microsoft-owned prodcuts, and the amount of documentation that exists to help get the environment setup. The rest of this document will assume that you're using VSCode as your IDE.The project itself is coded primarily in Python.
+#### Terraform State Storage Account
+The third and final storage account in the starter kit in the Terraform State Storage Account. This account is shared across all deployed environments of the starter kit (dev, prod, QA, etc..) and is used to store the Terraform state of each environment. For more information Terraform state please refer to the [Infrastructure as Code (IaC)](#infrastructure-as-code-iac) section of this doc.
```
Comment (nickclyde) [inline docs/getting_started.md:69]:
```suggestion
The third and final storage account in the starter kit is the Terraform State Storage Account. This account is shared across all deployed environments of the starter kit (dev, prod, QA, etc..) and is used to store the Terraform state of each environment. For more information Terraform state please refer to the [Infrastructure as Code (IaC)](#infrastructure-as-code-iac) section of this doc.
```

File: docs/getting_started.md (hunk 2 above)
Comment (nickclyde) [reply docs/getting_started.md:72]: Also `pipline` -> `pipeline`

**Review by nickclyde — CHANGES_REQUESTED** [review]
Thanks for writing this all up Dan! Just a few minor typo/punctuation notes.

File: docs/getting_started.md (hunk 1 above)
Comment (DanPaseltiner) [reply docs/getting_started.md]: removed.

**Review by nickclyde — APPROVED** [review]
Looks great!

**Pull request merged**